contracts/bond/Bond.sol

```diff
@@ -193,6 +193,13 @@ contract Bond is ERC20Upgradeable, OwnableUpgradeable, PausableUpgradeable {
         return _excessDebtTokens;
     }
 
+    /**
+     * @dev Whether or not the Bond contract has achieved full collateral target.
+     */
+    function hasFullCollateral() public view returns (bool) {
```

Comment:
question(non-blocking): maybe make this function and most other functions "virtual"? In that case it will allow to override it from BondV2, let's say, which can inherit most of the code. Also why not make _debtTokensRemaining() as "internal" rather than "private"?

Reply:
Good question.

Using `private` rather than `internal` is just tighter encapsulation. 
In the future if the scope needs widening to support inheritance of a BondV2, then the time to change the scope is then ...but until then, narrow as possible.

Rather than marking individual methods with virtual, I'd say that defining sensible interfaces would be the way to go ...but we're not there yet with the requirements.